integrations/google_genai/src/haystack_integrations/components/embedders/google_genai/document_embedder.py

```diff
@@ -0,0 +1,199 @@
+        """
+        Prepare the texts to embed by concatenating the Document text with the metadata fields to embed.
+        """
+        texts_to_embed = {}
+        for doc in documents:
+            meta_values_to_embed = [
+                str(doc.meta[key]) for key in self.meta_fields_to_embed if key in doc.meta and doc.meta[key] is not None
+            ]
+
+            texts_to_embed[doc.id] = (
+                self.prefix + self.embedding_separator.join([*meta_values_to_embed, doc.content or ""]) + self.suffix
+            )
+
+        return texts_to_embed
+
+    def _embed_batch(self, texts_to_embed: Dict[str, str], batch_size: int) -> Tuple[List[List[float]], Dict[str, Any]]:
+        """
+        Embed a list of texts in batches.
+        """
+
+        all_embeddings = []
+        meta: Dict[str, Any] = {}
+        for batch in tqdm(
+            batched(texts_to_embed.items(), batch_size), disable=not self.progress_bar, desc="Calculating embeddings"
+        ):
+            args: Dict[str, Any] = {"model": self.model, "contents": [b[1] for b in batch]}
+            if self.config:
+                args["config"] = types.EmbedContentConfig(**self.config) if self.config else None
+
+            try:
+                response = self.client.models.embed_content(**args)
+            except Exception as exc:
+                ids = ", ".join(b[0] for b in batch)
+                msg = "Failed embedding of documents {ids} caused by {exc}"
+                logger.exception(msg, ids=ids, exc=exc)
+                continue
+
+            embeddings = [el.values for el in response.embeddings]
+            all_embeddings.extend(embeddings)
+
+            if "model" not in meta:
+                meta["model"] = self.model
+
+        return all_embeddings, meta
+
+    @component.output_types(documents=List[Document], meta=Dict[str, Any])
+    def run(self, documents: List[Document]) -> Dict[str, Union[List[Document], Dict[str, Any]]]:
+        """
+        Embeds a list of documents.
+
+        :param documents:
+            A list of documents to embed.
+
+        :returns:
+            A dictionary with the following keys:
+            - `documents`: A list of documents with embeddings.
+            - `meta`: Information about the usage of the model.
+        """
+        if not isinstance(documents, list) or (documents and not isinstance(documents[0], Document)):
+            error_message_documents = (
+                "GoogleGenAIDocumentEmbedder expects a list of Documents as input. "
+                "In case you want to embed a string, please use the GoogleGenAITextEmbedder."
+            )
+            raise TypeError(error_message_documents)
+
+        texts_to_embed = self._prepare_texts_to_embed(documents=documents)
+
+        embeddings, meta = self._embed_batch(texts_to_embed=texts_to_embed, batch_size=self.batch_size)
+
+        for doc, emb in zip(documents, embeddings):
```

Comment:
There's a slight issue here. `texts_to_embed` is a Dict and `embeddings` is a list so this zip isn't guaranteed to match the order of embeddings to the correct documents. 

Since we are removing the try-except block in the run method, it'd be easiest to then make `texts_to_embed` a List of strings that have the same order as the original documents. You can see an example of this in our Cohere DocumentEmbedder integration [here](https://github.com/deepset-ai/haystack-core-integrations/blob/cfd52540ac277284eb7b3d3e735fe56eda3c33da/integrations/cohere/src/haystack_integrations/components/embedders/cohere/document_embedder.py#L135)